test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -0,0 +1,199 @@
+package docker_e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"celestiaorg/celestia-app/test/docker-e2e/dockerchain"
+
+	"github.com/celestiaorg/celestia-app/v5/pkg/appconsts"
+	"github.com/celestiaorg/tastora/framework/testutil/wait"
+)
+
+// TestCelestiaAppMinorUpgrade tests a simple upgrade from one minor version to another.
+func (s *CelestiaTestSuite) TestCelestiaAppMinorUpgrade() {
+	t := s.T()
+	if testing.Short() {
+		t.Skip("skipping upgrade test in short mode")
+	}
+
+	ctx := context.Background()
+	const (
+		baseVersion   = "v4.0.2-mocha"
+		targetVersion = "v4.0.7-mocha"
+	)
+
+	cfg := dockerchain.DefaultConfig(s.client, s.network)
+	cfg.Tag = baseVersion
+	builder := dockerchain.NewCelestiaChainBuilder(t, cfg)
+
+	chain, err := builder.Build(ctx)
+	s.Require().NoError(err)
+
+	err = chain.Start(ctx)
+	s.Require().NoError(err)
+
+	s.Require().NoError(wait.ForBlocks(ctx, 5, chain))
+
+	err = chain.Stop(ctx)
+	s.Require().NoError(err)
```

Comment:
maybe we can deploy tx sim here in between so it spams some transactions?

Reply:
I think a simple bank send would be enough to make sure it works. Do you think we need to have more tx?